File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        map<int, vector<int>> mi;
        for(int i = 0; i < nums.size(); i++){
            m[nums[i]]++;
            mi[nums[i]].push_back(i);
        }
        for(int i = 0; i < nums.size(); i++){
            m[nums[i]]--;
            if(m[target-nums[i]]>0 && target-nums[i] != nums[i]){
                return {mi[nums[i]][0], mi[target-nums[i]][0]};
            }else if(m[target-nums[i]]>0 && target-nums[i] == nums[i]){
                return {mi[nums[i]][0], mi[target-nums[i]][1]};
            }
            m[nums[i]]++;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        # loop through nums and store each element in hashmap
        for i in range(len(nums)):
            hashmap[nums[i]] = i
        # loop again and check if the value exists or not
        for i in range(len(nums)):
            if target - nums[i] in hashmap and hashmap[target-nums[i]] != i:
                return [i, hashmap[target-nums[i]]]